tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -427,35 +469,33 @@ static void *read_abort_setup(void)
 
 ZTEST_USER(uart_async_read_abort, test_read_abort)
 {
-	uint8_t rx_buf[100];
-	uint8_t tx_buf[100];
+	DATA_TYPE rx_buf[100];
+	DATA_TYPE tx_buf[100];
 
 	memset(rx_buf, 0, sizeof(rx_buf));
 	memset(tx_buf, 1, sizeof(tx_buf));
 
-	uart_rx_enable(uart_dev, rx_buf, sizeof(rx_buf), 50 * USEC_PER_MSEC);
+	UART_ENABLE_RX(uart_dev, rx_buf, ARRAY_SIZE(rx_buf), 50 * USEC_PER_MSEC);
 
-	uart_tx(uart_dev, tx_buf, 5, 100 * USEC_PER_MSEC);
+	UART_TX(uart_dev, tx_buf, 5, 100 * USEC_PER_MSEC);
 	zassert_equal(k_sem_take(&tx_done, K_MSEC(100)), 0, "TX_DONE timeout");
 	zassert_equal(k_sem_take(&rx_rdy, K_MSEC(100)), 0, "RX_RDY timeout");
-	zassert_equal(memcmp(tx_buf, rx_buf, 5), 0, "Buffers not equal");
-
-	uart_tx(uart_dev, tx_buf, 95, 100 * USEC_PER_MSEC);
+	zassert_equal(memcmp(tx_buf, rx_buf, 5 * sizeof(DATA_TYPE)), 0, "Buffers not equal");
 
-	k_timer_start(&read_abort_timer, K_USEC(300), K_NO_WAIT);
-
-	/* RX will be aborted from k_timer timeout */
+	UART_TX(uart_dev, tx_buf, 95, 100 * USEC_PER_MSEC);
 
+	/* RX will be aborted on the 2nd TX_DONE but no guarantees can be made on buf content
+	 * because of multithreading scheduling
+	 */
 	zassert_equal(k_sem_take(&tx_done, K_MSEC(100)), 0, "TX_DONE timeout");
 	zassert_equal(k_sem_take(&rx_disabled, K_MSEC(100)), 0,
 		      "RX_DISABLED timeout");
 	zassert_false(failed_in_isr, "Unexpected order of uart events");
-	zassert_not_equal(memcmp(tx_buf, rx_buf, 100), 0, "Buffers equal");
```

Comment:
@JvanDooren, currently the test checks that receive was actually aborted by making sure that tx_buf and rx_buf don't have the same contents. With the proposed change it would only check that uart_rx_disable(uart_dev) doesn't return an error. There is nothing to confirm that the receiving actually stopped. This makes the test weaker IMO. 
Does this test fail on your hardware without the change?

Reply:
It did fail, but somehow not anymore, reverted the change